### ISSUE TITLE
Bug: Parameter Server not stopping

### ISSUE THREAD

**Issue by l45k** [hypha-space/hypha#229]
## Description

The PS can't be stopped with ctrl+c.

## Symptoms

```log
2025-12-11T12:34:03.209430Z  INFO hypha_worker::executor::parameter_server: Executing parameter server job job_spec=JobSpec { job_id: aa6f9913-88cf-41ca-ab99-6dac8e15ee8e, executor: Aggregate(AggregateExecutor { descriptor: AggregateExecutorDescriptor { name: "parameter-server" }, config: AggregateExecutorConfig { optimizer: Nesterov { learning_rate: 0.7, momentum: 0.9 } } }) }
2025-12-11T12:34:03.209807Z  INFO hypha_worker::arbiter: Task successfully dispatched job_id=f9fe7b34-879e-4506-aa3a-dd6784df2b86 scheduler_id=12D3KooWJfDWPLaH5Nh2qkD4JJkTYmAyDeNtZVtuo7LkTXRJfPce
^C2025-12-11T12:35:15.162850Z  INFO hypha_worker: Received SIGINT, shutting down
2025-12-11T12:35:19.918926Z  INFO hypha_worker::arbiter: Dropped 1 expired leases expired=[Lease { id: 47afcfa8-8fee-47e4-9148-1df64ce1c0f3, leasable: ResourceLease { peer_id: PeerId("12D3KooWJfDWPLaH5Nh2qkD4JJkTYmAyDeNtZVtuo7LkTXRJfPce"), reservation: 5c4b7929-cccc-4d48-b6b0-9633bbc2a1a8 }, timeout: SystemTime { tv_sec: 1765456519, tv_nsec: 912519000 } }]
^C2025-12-11T12:35:22.359005Z ERROR hypha_network::request_response: Outbound request-response failure peer=12D3KooWJfDWPLaH5Nh2qkD4JJkTYmAyDeNtZVtuo7LkTXRJfPce request_id=OutboundRequestId(137) error=DialFailure
2025-12-11T12:35:22.498829Z ERROR hypha_network::request_response: Outbound request-response failure peer=12D3KooWJfDWPLaH5Nh2qkD4JJkTYmAyDeNtZVtuo7LkTXRJfPce request_id=OutboundRequestId(138) error=DialFailure
2025-12-11T12:35:23.391545Z ERROR hypha_network::request_response: Outbound request-response failure peer=12D3KooWJfDWPLaH5Nh2qkD4JJkTYmAyDeNtZVtuo7LkTXRJfPce request_id=OutboundRequestId(139) error=DialFailure
^C^C^C
```

This is a flaky error.

Feel free to share whatever helps us see what you see! 

IMPORTANT: Remember Github issues are public, so please do not include any sensitive information!
-->

**Comment by nfnt** [comment]
I can reproduce this by stopping the scheduler process while a job is running. Once the scheduler is stopped, the parameter server will hang after receiving SIGINT.
